### ISSUE TITLE
Preserve order of class properties in serialize

### ISSUE THREAD

**Issue by dahoward** [cbeust/klaxon#186]
Would like to be able to preserve the order of the class properties when serializing.  For example, the serialized output of the following code is shown below:
    class Person(val name: String, val age: Int)
    val aPerson = Person("Bobby", 2)
    println(Klaxon().toJsonString(aPerson))
==> {"age" : 2, "name" : "Bobby"}

Klaxon serialization alphabetized the properties.  For ease of human reading, it can be preferable to order the Json attributes in a different way other than alphabetical such as by importance or by a logical grouping.  Is that a feature that can be provided?  Thanks.

**Comment by cbeust** [comment]
I'm not sure it's possible since Kotlin reflection doesn't appear to preserve that ordering:

```
        class Person(val name: String, val age: Int)
        println(Person::class.memberProperties)
        // prints
        // val com.beust.klaxon.`KlaxonTest$alphabetize$Person`.age: kotlin.Int
        // val com.beust.klaxon.`KlaxonTest$alphabetize$Person`.name: kotlin.String
```

Feel free to reopen if you find a way...

**Comment by cbeust** [comment]
@StephenOTT I'm open to the idea. I have a few open questions about the design, though: should this be a class level annotation (like Jackson):

```
@JsonPropertyOrder("id", "name")
class Data
```

Or specified at the property level:

```
class Data(
  @JsonPropertyOrder(index = 0) val name: String
 @JsonPropertyOrder(index = 1) val id: String
)
```

?

The advantage of the first approach is that it's one annotation in one place, but the downside is that a refactoring or a change in the JSON payload names can break it.

The second approach is a bit more tedious but will never break.

Thoughts?

**Comment by StephenOTT** [comment]
Option two would be great.  Do you see this as a codegen or runtime reflection calculation?  Potentially a codegen approach to generate the order so you don't need to sort every time?

**Comment by cbeust** [comment]
Runtime, the time to sort these keys is meaningless. Any more votes in favor of either option?

**Comment by cbeust** [comment]
I went with the `@Json(index = ...)` approach.

Documentation:  https://github.com/cbeust/klaxon#index

**Comment by cbeust** [comment]
Available in 5.1.
